text/go/0001-go-bom.md

```diff
@@ -0,0 +1,163 @@
+
+Add support for the generation of the Bill of Materials in the Go Language Family in order to provide entries for
+modules installed during the build process that can be found on the final app image.
+
+## Motivation
+
+As a part of an initiative to provide a Bill of Materials (BOM) for all the Paketo Language Family Buildpacks, in
+previous months
+the [Node.JS BOM Module](https://github.com/paketo-buildpacks/rfcs/blob/main/text/nodejs/0013-install-bom.md) was
+developed.
+
+We would like to add support for BOM generation in the Go language family. We would like to create a new buildpack to
+provide this. We do not anticipate the need for changes to other buildpacks in the language family.
+
+We propose to support generating BOMs only for projects which utilize vendoring via `go mod`. At this time we do not
+propose to support any other vendoring solutions
+(e.g. `dep` or files copied directly into the `vendor` directory without a
+`go.mod` file). Extending the new buildpack to support to other vendoring solutions could be a future RFC, if we
+discover a clear requirement for this.
+
+## Implementation
+
+### Tool details
+
+The Bill of Materials that will result should be an accurate accounting of all the modules that are defined and
+installed by `go modules`.
+
+There already exists a tool called [CycloneDX Gomod](https://github.com/CycloneDX/cyclonedx-gomod) which is an official
+CycloneDX supported-tool that creates valid CycloneDX BOMs for Go applications. CycloneDX is a BOM format supported by
+the OWASP Foundation, and is a serious contender (at the time of this RFC creation) as a format we will eventually
+support.
+
+The CycloneDX Go Module tool is available as a pre-built binary for multiple platforms, and also it can easily be
+installed via `go install github.com/CycloneDX/cyclonedx-gomod`.
+
+This tool supports three different cases of BOM generation:
+
+1. `app`: Include only those modules that the target application actually depends on. Modules required by tests or
+   packages that are not imported by the application are not included. Build constraints are evaluated, which enables a
+   very detailed view of what's really compiled into an application's binary.
+2. `mod`: Include the aggregate of modules required by all packages in the target module. This optionally includes
+   modules required by tests and test packages. Build constraints are NOT evaluated, allowing for a "whole picture" view
+   on the target module's dependencies.
+3. `bin`: Offers support of generating rudimentary SBOMs from binaries built with Go modules.
+
+Of these, we are only interested in `app`. The buildpacks do not currently support building modules, or libraries, which
+is the primary use-case for `mod`. The `bin` command is for pre-built executables, which is unnecessary for the
+buildpack as it has access to the source code. The BOM created by the `bin` command also contains less information than
+the one created via the `app` command.
```

Comment:
Can you explain more why we are not interested in the `mod` case? Why wouldn't we want to include as much information about the dependency tree (and anything that comes into contact with the app) as possible?